crates/turbopack/src/module_options/mod.rs

```diff
@@ -197,21 +197,24 @@ impl ModuleOptionsVc {
                     ModuleRuleCondition::ResourcePathEndsWith(".js".to_string()),
                     ModuleRuleCondition::ResourcePathEndsWith(".jsx".to_string()),
                 ]),
-                vec![ModuleRuleEffect::ModuleType(ModuleType::Ecmascript(
-                    app_transforms,
-                ))],
+                vec![ModuleRuleEffect::ModuleType(ModuleType::Ecmascript {
+                    transforms: app_transforms,
+                    options: Default::default(),
```

Comment:
You probably want to get the options from a new field `enable_tree_shaking` in the `ModuleOptionsContext` to allow enabling tree-shaking from outside.